server/workflow/workflow_server.go

```diff
@@ -40,14 +54,21 @@ func (s *workflowServer) CreateWorkflow(ctx context.Context, req *workflowpkg.Wo
 	if req.Workflow.Namespace == "" {
 		req.Workflow.Namespace = req.Namespace
 	}
-
-	if req.InstanceID != "" {
-		labels := req.Workflow.GetLabels()
-		if labels == nil {
-			labels = make(map[string]string)
+	if s.mode == GRPCServerMode {
```

Comment:
should also work for kube client - should not need the guard condition

Reply:
If the `mode` is really needed, adding this makes it better, because the label has been added somewhere else. But I'm also ok to remove it.